File: src/main/java/api/routes/locale/LocaleController.java
package api.routes.locale;

import api.routes.locale.dto.LocaleDto;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/locale")
@RequiredArgsConstructor
public class LocaleController {

    private final LocaleService localeService;

    @PostMapping("")
    public ResponseEntity<LocaleResponse> setLocale(@Valid @RequestBody LocaleDto localeDto) {
        return localeService.setLocale(localeDto);
    }
}

File: src/main/kotlin/api/routes/locale/LocaleController.kt
package api.routes.locale

import api.plugins.routeWithJwt
import api.utils.respond
import io.ktor.server.application.*
import io.ktor.server.request.*
import io.ktor.server.routing.*
import net.dv8tion.jda.api.sharding.ShardManager
import org.koin.ktor.ext.inject

fun Routing.locale() {
    val shardManager: ShardManager by inject()
    val service = LocaleService(shardManager)

    routeWithJwt("/locale") {
        post {
            val dto = call.receive<LocaleDto>()
            call.respond(service.setLocale(dto))
        }
    }
}